prow/pod-utilities.md

```diff
@@ -20,26 +20,24 @@ to the user-provided `PodSpec`, as well as by overwriting the `Container` entryp
 
 ## Writing a ProwJob that uses Pod Utilities
 
-Writing a ProwJob that uses the Pod Utilites is much easier than writing one
-that doesn't because the Pod Utilities will transparently handle many of the
-tasks the job would otherwise need to do in order to prepare its environment
-and output more than pass/fail. Historically, this was achieved by wrapping
-every job with a [bootstrap.py](jenkins/bootstrap.py) script that handled cloning
-source code, preparing the test environment, and uploading job metadata, logs,
-and artifacts. This was cumbersome to configure and required every job to be
-wrapped with the script in the job image. The pod utilites achieve the same goals
-with less configuration and much simpler job images that are easier to develop
-and less coupled to Prow.
+### What the test container can expect
 
-### What to expect
+Example test container script:
+```bash
+pwd # my repo root
+ls path/to/file/in/my/repo.txt # access repo file
+ls ../other-repo # access repo file in another repo
+echo results.txt > $ARTIFACTS # result info that will be uploaded to GCS.
+# logs, and job metadata are automatically uploaded.
```

Comment:
There ought to be a way for me to inject user metadata, how does that work, if at all?

Reply:
If a job has additional metadata it would just dump it to the $ARTIFACTS dir for upload. There isn't currently any way to distinguish custom metadata from other custom artifacts AFAIK.

Reply:
Doesn't it have to be in a special metadata.json file? If so that might be useful to document?

Reply:
The pod utilities don't support populating the `metadata` field in finished.json yet. We will want to add that before migrating too many jobs in order to allow for custom metadata to be specified.

I'll make an issue for that. I don't think we need to add anything to the docs about this yet.

Reply:
#8104